src/wallet/db.h

```diff
@@ -68,7 +68,7 @@ class CDBEnv
     typedef std::pair<std::vector<unsigned char>, std::vector<unsigned char> > KeyValPair;
     bool Salvage(const std::string& strFile, bool fAggressive, std::vector<KeyValPair>& vResult);
 
-    bool Open(const fs::path& path);
+    bool Open(const fs::path& path, bool retry = 0);
```

Comment:
`= false`?

Reply:
BTW, how about:
```cpp
private:
    bool Open(const fs::path& path, bool retry);

public:
    bool Open(const fs::path& path) { return Open(path, true); }
```
this way the *public* API remains, for instance, https://github.com/bitcoin/bitcoin/pull/11904/files#diff-5cd083d2c34da5f0ac380011f7b725d2R312 goes away.

Reply:
I'd prefer to leave it explicit I think, is there any other benefit to hiding the retry parameter?

Reply:
It's an internal detail, you should not be able to call `Open(path, false)`, it's only there for the 2nd attempt.

Reply:
Its also called without retry here: 
https://github.com/MeshCollider/bitcoin/blob/f70ce89806a7f0ce1755e7d074f6e07753a85694/src/wallet/db.cpp#L436

Reply:
Ok, missed that.